setup.py

```diff
@@ -11,7 +18,7 @@
     keywords="genetic programming",
     url="https://github.com/jakobj/python-gp",
     python_requires=">=3.6, <4",
-    install_requires=["sympy", "torch", "numpy"],
+    install_requires=read_requirements(),
```

Comment:
You can specify the extra requirements in a separate entry, like so:

```python
    extras_require={
        'sympy': ["sympy"]
        'torch': ["torch"],
        'numpy': ["numpy"]
    },
```

and then one can install them via pip by saying:
`pip install python-gp[sympy, torch]` or locally
`pip install .[sympy, torch]`

(Note that if you're using zsh as your shell you need to escape the whole thing to make it work because square brackets are protected in zsh: e.g. `pip install '.[sympy]'`

Reply:
thanks, that's a good idea. however, i think `pip install -r optional-requirements.txt` is even simpler and a separate file with a list of modules easier to maintain. what do you think?

Reply:
The nice thing about what I suggested is that the user can switch off the extra requirements in a targeted manner. For instance, you might want torch but not sympy.

Reply:
:+1: